lib/socket.js

```diff
@@ -84,7 +85,7 @@ class Transport extends EventEmitter {
 
 const socketFactory = (connect) => {
   const resultConnectFn = (...options) => {
-    const callback = common.extractCallback(options);
+    const callback = common.cbExtract(options);
```

Comment:
@tshemsedinov, but it's a completely natural behavior in Node.js for `TypeError` to occur whenever you provide invalid arguments to the function (which is the case with providing no callback to the `connect` function).
This means that the user gets `TypeError` thrown only when he/she is using the library incorrectly.